staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -113,6 +113,20 @@ type Config struct {
 	// Version forces a specific version to be used (if registered)
 	// Do we need this?
 	// Version string
+
+	// BearerTokenSource for managing the token
+	BearerTokenSource BearerTokenSource
```

Comment:
@liggitt  if `clientcmdapi.AuthProviderConfig` exits in the current context, the `BearerTokenSource` will be set.